### PR TITLE
feat(NODE-5858): Add support for older bson versions

### PR DESCRIPTION
**Pull request by W-A-James**
Supports testing against bson versions 1 and 2

### CODE REVIEW

File: packages/bson-bench/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   "author": "The MongoDB NodeJS Team <dbx-node@mongodb.com>",
   "license": "Apache-2.0",
   "dependencies": {
-    "bson": "^6.1.0"
+    "bson": "4.7"
```
Comment (W-A-James) [inline packages/bson-bench/package.json:20]: Note: this change was meant to be merged into main prior to this, but was used when collecting results for NODE-4004.

Comment (dariakp) [reply packages/bson-bench/package.json:20]: That's confusing, what's the bson library package used for inside the tooling?

Comment (dariakp) [reply packages/bson-bench/package.json:20]: More to the point, how can this tooling be cross version and cross library compatible if it depends on a specific library at a specific version?

Comment (nbbeeken) [reply packages/bson-bench/package.json:20]: Isn't this the "setup" bson used to generate docs and such? why wouldn't this be the latest?

File: packages/bson-bench/src/base.ts [hunk 2]
```diff
@@ -28,14 +29,17 @@ function reportErrorAndQuit(error: Error) {
...
+function run(bson: BSONLib | ConstructibleBSON, config: BenchmarkSpecification) {
   let fn:
     | ((b: Uint8Array, options?: any) => any)
     | ((o: any, options?: any) => Uint8Array)
     | undefined = undefined;
   let documentSizeBytes: number;
   let doc: any;
 
+  // Check if this is bson < v4
+  if (!('serialize' in bson)) bson = new bson();
```
Comment (nbbeeken) [inline packages/bson-bench/src/base.ts:41]: nit, don't feel strongly, but "typeof of function" would make sense when the incoming type is either BSON object or BSON constructor

File: packages/bson-bench/package.json (hunk 1 above)
Comment (W-A-James) [reply packages/bson-bench/package.json:20]: This bson version needs to be v4 for compatibility with bson-ext. It's only used [here](https://github.com/mongodb-js/dbx-js-tools/blob/c9c18764d3ec4b462d198bc6312cf6b707809811/packages/bson-bench/src/base.ts#L44-L45) in the event that we don't have access to EJSON to parse our test documents to an object as is the case in bson-ext. bson-ext only works with documents created with bson v4.

Comment (dariakp) [reply packages/bson-bench/package.json:20]: If that's the case, can we add a comment to that extent in the line you linked for clarity?

Comment (dariakp) [reply packages/bson-bench/package.json:20]: Thanks, all good on my end

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**